### PR TITLE
Split credentials for different providers

### LINKED ISSUE #15: Separate queue vs s3 configs
Be able to use different vendors (AWS vs B2) for storage configs. Right now there is a single AWS config that is tightly coupled to both the queue and storage.

### PR DESCRIPTION
**Pull request by chumaumenze**
Use different AWS configs for AWS-specific services and other S3-compatible service providers. Force usage of S3 URL path style

Fixes #15

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	client.storageConfig = aws.NewConfig().
+		WithRegion(c.Storage.Region).
+		WithEndpoint(c.Storage.Endpoint).
+		WithCredentials(storageCreds).
+		WithS3ForcePathStyle(true)
+
+	return &client
+}
+
+func (c *Client) S3() *s3.S3 {
```
Comment (poundifdef) [inline client/client.go:39]: Is this thread safe?

Comment (chumaumenze) [reply client/client.go:39]: Yes, this is safe.

Comment (chumaumenze) [reply client/client.go]: Yes, this is not safe.

**Comment by poundifdef** [comment]
Hi - I've tested this locally and everything is looking good. Thank you!

**Pull request merged**